apps/emqx_durable_storage/src/emqx_ds.erl

```diff
@@ -142,6 +144,8 @@
 
 -type delete_next_result() :: delete_next_result(delete_iterator()).
 
+-type error(Reason) :: {error, recoverable | unrecoverable, Reason}.
```

Comment:
Not sure if we want to invent a new boolean.

Reply:
IMO, having these constructors is better and more semantic, to avoid boolean blindness.

Reply:
My reasoning as well, those are usually more apparent and obvious in logs / stacktraces / tests. Also somewhat less awkward to refine or expand the set of classes in the future (which is, admittedly, unlikely).

Reply:
Not a strong opinion, but straight booleans work with constructs like `or`, `orelse`, ... Also they lead to nicer guards.

Reply:
Out of the box, true.  I think we have the option to create a semantic guard for that, similar to `?is_replay_ongoing`, that uses those constructs.